webots_ros2_epuck/EPUCK_ROS2.md

```diff
@@ -0,0 +1,206 @@
+# ROS2 for E-Puck
+Here you will find instructions on how to use utilize e-puck ROS2 API. 
+
+This documentation is common for for physical ([`epuck_ros2`](https://github.com/cyberbotics/epuck_ros2)) and simulated ([`webots_ros2_epuck`](https://github.com/cyberbotics/webots_ros2/tree/master/webots_ros2_epuck)) robot, so there is a slight difference in using the launch file.
+Launch file `example_launch.py` runs the Webots simulation and includes `example_tools_launch.py`.
+Therefore, you can use the launch file `example_tools_launch.py` for the physical robot with the same parameters as you would `example_launch.py` for the simulated robot.
+To emphasis this, you will see `[tools_]` in the commands which indicates you can use `example_tools_launch.py` and `example_launch.py` depending on whether you work with the physical or simulated robot, e.g.:
```

Comment:
I am not sure to understand why the name of the launch files are different, since the package names are not the same there should be no name clash if the launch files use the same names no?

Reply:
First, you have to run driver:
- `ros2 launch webots_ros2_epuck example_launch.py` for simulated robot or
- `ros2 launch epuck_ros2 example_launch.py` for physical robot.

Second, you can run extra tools/demos with:
`ros2 launch webots_ros2_epuck example_tools_launch.py`
(this works for both, simulated and physical robot)

For convenience, `webots_ros2_epuck/example_launch.py` also includes `webots_ros2_epuck/example_tools_launch.py` so you don't need to run two launch files.

But since you didn't understand it is either poorly explained or organized. Do you have any idea how to simplify?

Reply:
> First, you have to run driver:
> 
> * `ros2 launch webots_ros2_epuck example_launch.py` for simulated robot or
> * `ros2 launch epuck_ros2 example_launch.py` for physical robot.

Ok perfect, this is indeed what I would expect that behaviour and name of launch files are exactly the same for real and simulated robot (juste the package name changes).

> Second, you can run extra tools/demos with:
> `ros2 launch webots_ros2_epuck example_tools_launch.py`
> (this works for both, simulated and physical robot)
> 
> For convenience, `webots_ros2_epuck/example_launch.py` also includes `webots_ros2_epuck/example_tools_launch.py` so you don't need to run two launch files.
> 
> But since you didn't understand it is either poorly explained or organized. Do you have any idea how to simplify?

Ok, after reading it several times I understand know the difference, the difference is that example_launch.py doesn't not start the tools on the real robot but on the simulated one yes, am I right?

I think this is the perturbing point, I am wondering if we should not make the `example_launch.py` work and behave the exact same way with real robot and in simulation. And add an extra launch (e.g. `example_with_tools_launch.py` that will do what the current ``example_launch.py`` is doing in simulation). What do you think?

Reply:
I guess it is a good idea, we can proceed that!

Reply:
I think it is clear now. I also renamed `example_launch.py` to `robot_launch.py` to be closer to turtlebot convention

Reply:
Much clearer indeed, thank you!